frontend/packages/console-telemetry-plugin/src/listeners/segment.ts

```diff
@@ -0,0 +1,109 @@
+/* eslint-disable no-console */
+import { SEGMENT_API_KEY, TELEMETRY_DISABLED } from './const';
+
+const initSegment = () => {
+  // eslint-disable-next-line no-multi-assign
+  const analytics = ((window as any).analytics = (window as any).analytics || []);
+  if (analytics.initialize) {
+    return;
+  }
+  if (analytics.invoked) {
+    console.error('console-telemetry-plugin: segment snippet included twice');
+  } else {
+    analytics.invoked = true;
+    analytics.methods = [
+      'trackSubmit',
+      'trackClick',
+      'trackLink',
+      'trackForm',
+      'pageview',
+      'identify',
+      'reset',
+      'group',
+      'track',
+      'ready',
+      'alias',
+      'debug',
+      'page',
+      'once',
+      'off',
+      'on',
+      'addSourceMiddleware',
+      'addIntegrationMiddleware',
+      'setAnonymousId',
+      'addDestinationMiddleware',
+    ];
+    analytics.factory = function(e: string) {
+      return function() {
+        // eslint-disable-next-line prefer-rest-params
+        const t = Array.prototype.slice.call(arguments);
+        t.unshift(e);
+        analytics.push(t);
+        return analytics;
+      };
+    };
+    for (const key of analytics.methods) {
+      analytics[key] = analytics.factory(key);
+    }
+    analytics.load = function(key: string, e: Event) {
+      const t = document.createElement('script');
+      t.type = 'text/javascript';
+      t.async = true;
+      t.src = `https://cdn.segment.com/analytics.js/v1/${encodeURIComponent(key)}/analytics.min.js`;
+      const n = document.getElementsByTagName('script')[0];
+      if (n.parentNode) {
+        n.parentNode.insertBefore(t, n);
+      }
+      // eslint-disable-next-line no-underscore-dangle
+      analytics._loadOptions = e;
+    };
+    analytics.SNIPPET_VERSION = '4.13.1';
+    if (SEGMENT_API_KEY) {
+      analytics.load(SEGMENT_API_KEY);
+    }
```

Comment:
We call this method only when API key is defined, right?

```suggestion
    analytics.load(SEGMENT_API_KEY);
```